.travis/build.sh

```diff
@@ -1,48 +1,47 @@
 #!/bin/bash
+set -e
+set -x
 
-export CXX=$COMPILER
-
-# just to make sure
-git submodule update --init --recursive
+LOCAL_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+ROOT_DIR=$(dirname "$LOCAL_DIR")
+cd "$ROOT_DIR"
 
 mkdir build
 cd build
 
-if [[ $BUILD_TARGET == 'android' ]]; then
-#***************#
-# Android build #
-#***************#
-  export ANDROID_NDK=/opt/android_ndk
-  sh ../scripts/build_android.sh
-elif [[ $BUILD_TARGET == 'ios' ]]; then
-#***************#
-# iOS build     #
-#***************#
-  # Note: we will only build arm64 in the travis case for faster compilation.
-  # You might want to build a fat binary containng armv7, armv7s and arm64.
-  # This can be done by simply not passing in the CMAKE_OSX_ARCHITECTURES flag.
-  sh ../scripts/build_ios.sh -DCMAKE_OSX_ARCHITECTURES=arm64
-elif [[ $TRAVIS_OS_NAME == 'osx' ]]; then
-#************#
-# OS X build #
-#************#
-  cmake .. \
-      -DCMAKE_VERBOSE_MAKEFILE=ON \
-      -DUSE_OPENCV=OFF \
-  && make
+# Special cases - run script and exit
+if [ "$BUILD_ANDROID" = 'true' ]; then
+    export ANDROID_NDK=/opt/android_ndk
+    sh "${ROOT_DIR}/scripts/build_android.sh"
+    exit 0
+fi
+if [ "$BUILD_IOS" = 'true' ]; then
+    sh "${ROOT_DIR}/scripts/build_ios.sh" -DCMAKE_OSX_ARCHITECTURES=arm64
+    exit 0
+fi
+
+# Configure
+CMAKE_ARGS=('-DCMAKE_VERBOSE_MAKEFILE=ON')
+if [ "$TRAVIS_OS_NAME" = 'osx' ]; then
+    CMAKE_ARGS+=('-DUSE_OPENCV=OFF')
```

Comment:
Is this actually necessary? I know it was in the old build but it might be good to cover OpenCV on macOS